core/internal/flux_monitor_simulated_blockchain_test.go

```diff
@@ -0,0 +1,370 @@
+	subscription, err := fa.aggregatorContract.WatchSubmissionReceived(nil,
+		submissionReceived, []*big.Int{}, []uint32{},
+		[]common.Address{fa.nallory.From})
+	require.NoError(t, err, "failed to subscribe to SubmissionReceived events")
+	defer subscription.Unsubscribe()
+
+	// Create FM Job, and wait for job run to start (the above UpdateAnswer call
+	// to FluxAggregator contract initiates a run.)
+	buffer := cltest.MustReadFile(t, "testdata/flux_monitor_job.json")
+	var job models.JobSpec
+	require.NoError(t, json.Unmarshal(buffer, &job))
+	initr := &job.Initiators[0]
+	initr.InitiatorParams.Feeds = cltest.JSONFromString(t, fmt.Sprintf(`["%s"]`,
+		mockServer.URL))
+	initr.InitiatorParams.PollingInterval =
+		models.Duration(timeout)
+	initr.InitiatorParams.Address = fa.aggregatorContractAddress
+	j := cltest.CreateJobSpecViaWeb(t, app, job)
+	jrs := cltest.WaitForRuns(t, j, app.Store, 1) // Submit answer from
+	reportedPrice := jrs[0].RunRequest.RequestParams.Get("result").String()
+	assert.Equal(t, reportedPrice, fmt.Sprintf("%d", reportPrice),
+		"failed to report correct price to contract")
+	var receiptBlock uint64
+	select { // block until FluxAggregator contract acknowledges chainlink message
+	case log := <-submissionReceived:
+		receiptBlock = log.Raw.BlockNumber
+	case <-time.After(timeout):
+		t.Fatalf("chainlink failed to submit answer to FluxAggregator contract")
+	}
+	checkUpdateAnswer(t, answerParams{fa: &fa, roundId: roundId,
+		answer: processedAnswer, from: fa.nallory, isNewRound: false,
+		completesAnswer: true}, initialBalance, receiptBlock)
+
+	//- have the malicious node start the next round.
+	nextRoundBalance := initialBalance - fee
+	// Triggers a new round, since price deviation exceeds threshold
+	reportPrice = answer + 1
+	select {
+	case log := <-submissionReceived:
+		receiptBlock = log.Raw.BlockNumber
+	case <-time.After(2 * timeout):
+		t.Fatalf("chainlink failed to submit answer to FluxAggregator contract")
+	}
+	newRound := roundId + 1
+	processedAnswer = 100 * reportPrice
+	checkUpdateAnswer(t, answerParams{fa: &fa, roundId: newRound,
+		answer: processedAnswer, from: fa.nallory, isNewRound: true,
+		completesAnswer: false}, nextRoundBalance, receiptBlock)
+	//- successfully close the round through the submissions of the other nodes
+	updateAnswer(t, answerParams{fa: &fa, roundId: newRound,
+		answer: processedAnswer, from: fa.neil, isNewRound: false,
+		completesAnswer: true})
+
+	//- have the malicious node try to start another round repeatedly until the
+	//roundDelay is reached, making sure that it isn't successful
+	// Triggers a new round, since price deviation exceeds threshold
+	reportPrice = answer + 1
+	select {
+	case <-submissionReceived:
+		t.Fatalf("chainlink node updated FA, even though it's not allowed to")
+	case <-time.After(5 * timeout):
+	}
+	// Could add a check for "not eligible to submit here", using the memory log
```

Comment:
This sounds like a good idea -- want to make sure it's failing to submit for the right reason.

Reply:
Added. (I was able to do this with `gomega.Eventually`.)